### ISSUE TITLE
splitLongitude produces NaN texture coordinates

### ISSUE THREAD

**Issue by OmarShehata** [AnalyticalGraphicsInc/cesium#7406]
```js
worldRectangle = scene.primitives.add(new Cesium.Primitive({
        geometryInstances : new Cesium.GeometryInstance({
            geometry : new Cesium.RectangleGeometry({
                rectangle : Cesium.Rectangle.fromDegrees(-180.0, -90.0, 180.0, 90.0),
                vertexFormat : Cesium.EllipsoidSurfaceAppearance.VERTEX_FORMAT
            })
        }),
        appearance : new Cesium.EllipsoidSurfaceAppearance({
            aboveGround : false
        }),
        show : false
    }));
```

I've discovered that the texture coordinates attribute has NaN's in it. If I put the following check in `Source/Core/AttributeCompression.js` everything works fine:

https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/AttributeCompression.js#L308-L316

```
if (isNaN(textureCoordinates.x) || isNaN(textureCoordinates.y)) {
       return 0;
}
```

**Comment by mramato** [comment]
May not be the same problem.  I can run the tests, but the Materials demo crashes.

**Comment by likangning93** [comment]
Looks like under linux it just kind of freezes after a few frames.

**Comment by alx696** [comment]
Ubuntu 18.04 Chrome71 crash after the Globe show.

**Comment by mramato** [comment]
@lilleyse @bagnell this is a huge issue that we need to address ASAP.  Who would be the best person to look at this?

**Comment by lilleyse** [comment]
@OmarShehata is going to look into this a bit today. It should be easy to identify which geometry or material is causing the problem in this demo.

**Comment by OmarShehata** [comment]
Preliminary results: I'm not sure why yet, but it seems to be having trouble with `worldRectangle`. 

```
worldRectangle = scene.primitives.add(new Cesium.Primitive({
        geometryInstances : new Cesium.GeometryInstance({
            geometry : new Cesium.RectangleGeometry({
                rectangle : Cesium.Rectangle.fromDegrees(-180.0, -90.0, 180.0, 90.0),
                vertexFormat : Cesium.EllipsoidSurfaceAppearance.VERTEX_FORMAT
            })
        }),
        appearance : new Cesium.EllipsoidSurfaceAppearance({
            aboveGround : false
        }),
        show : false
    }));
```

If I change the line:

```
rectangle : Cesium.Rectangle.fromDegrees(-180.0, -90.0, 180.0, 90.0),
```

To

```
rectangle : Cesium.Rectangle.fromDegrees(-180.0, -80.0, 180.0, 80.0),
```

It works fine.

**Comment by OmarShehata** [comment]
The other way to fix it is to comment out the appearance, so the following works:

```
worldRectangle = scene.primitives.add(new Cesium.Primitive({
        geometryInstances : new Cesium.GeometryInstance({
            geometry : new Cesium.RectangleGeometry({
                rectangle : Cesium.Rectangle.fromDegrees(-180.0, -90.0, 180.0, 90.0),
                vertexFormat : Cesium.EllipsoidSurfaceAppearance.VERTEX_FORMAT
            })
        }),
        //appearance : new Cesium.EllipsoidSurfaceAppearance({
        //    aboveGround : false
        //}),
        show : false
    }));
```

I've discovered that the texture coordinates attribute has NaN's in it. If I put the following check in `Source/Core/AttributeCompression.js` everything works fine:

https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/AttributeCompression.js#L308-L316

```
if (isNaN(textureCoordinates.x) || isNaN(textureCoordinates.y)) {
       return 0;
}
```

For some reason, this `st` attribute list contains NaN values:

https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/GeometryPipeline.js#L1403-L1412

**Comment by OmarShehata** [comment]
The issue seems specifically to be in the `compressVertices` step. Another way to make this Sandcastle work is simply to pass `compressVertices : false` to the `worldPrimitive`.

**Comment by hpinkos** [comment]
Cool, texture coordinate NaNs is a very fixable problem.  I guess this is something previous versions of Chrome tolerated but this latest release does not.

**Comment by OmarShehata** [comment]
Yup! That's the good news. I was worried it was some very specific obscure Chrome 71 WebGL issue, but you can see the NaNs are present in Firefox as well, and it just doesn't crash there.

**Comment by hpinkos** [comment]
I can probably take a look at this next week if no one else gets to it first.  The texture coordinate calculation happens here: https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/RectangleGeometryLibrary.js#L56-L68

**Comment by OmarShehata** [comment]
I was wrong about compressVertices. It gets NaN's before that, it's just that it only crashes during that step. 

@hpinkos I just checked RectangleGeometryLibrary and the produces coordinates there are clean.

The NaNs are introduced in `GeometryPipeline.splitLongitude(instances[i]);` Commenting out this line (or just running [the Sandcastle with scene3DOnly](https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/#c=7VyNT+M4Fv9XIqRT0jtIKdx87AKjY4AZjTTsIGCZlZbVKCRu6xvHztkO0D3xv6/tfNROnJImbWc0d5GgJLHfe7/n52f7vVfuA+pQEPIATxA4uMX34v6BUBRdVh4mBM0QxPL2Fo9THHJIsBMkCZodo2QanAfJecABhQHyEgpjyOE92HZYCDAYOP+9xY64rgIchQHjCPgRCFFAgWelMDhwnOHQeUeoM4WTKRI/HOKJA7FOIqNZMvMFKRDQAIfAj3NCzpGDwYNzAhhMY78UMBdHXuPgjsLQ+dnRnsmroMDqr+QVZBLnXKxN5MVnCRBv3QKgu21vl2I4JjRmzZTkBeNgosj5/lD9zYYZsi8fyYR8OSGIUP/fyaSJi7zCaYAxkCK71LU3e6o/rjx6qjAISZwQDDBvABDB8ThlSvR7EO57I393YBNSaVW2MtTrq7uKsJpE+Z9PA2GbT3XzfJvGSS/rrBCQxvmdW2eu7/b2eVp0WL2F3gntxUHiJ3iyhL1ZpLhTw9AWUT5oa4WzYJoxTgGe8KnovOu//gbzrGICfn5vk5klIEyFuStZd0eWFlioSind1QfBzx53npgnUxB+BfSOBDTqPjttVFa5flRmpz+mJL4WRua5Ome3CaP0yT3A6d0Fqg2Akhyb0Qj7Y5CDPogqJJQ73ZQ/nftU6/x53isoL2A4BCEEn8opNPq6aO2dAgnrHQQocur95bTK+7dzhuUC0ChpNlOThf6tpbNs7zAbnObzGBf4xCWWCEZSGgrediJu+Gf8pbQVeTMBvLQS/eUHnKTcMe6EYbvOP5oFN4lrFpnzOQXjIEVzdibxg8W0x4gEPDehmwClQNDl4JGnFOydepppbZtC+4wPfPoM8biyRAjaMXz01EZt19/bFovCS/lrb7DtFLs3ef+ieKjJNWjLK9vpHTmSmLOjgVEWY4UxcP4uGL5qyyFfs46KiZA/aNu9XBSPxFoOEqGJUR2pdFtXRcOYpJgHwmlxkrBWLKYQixMVY4LHa3+3JDgNaALw3BcuoEWBUBsutdUMzn1asEw3ePtTwrv7ea3zSjfMzSuX4Ni0bp2am6EOaEwC38MR4Pmt+2KX3fow2XV7dxZDxsRdr7OXhcj/z18riBD0WmZBOSht8WnDuHKAobYNf+5YRoEwAv78HuZRNBltL24zU+elFyvZy3yLY18xirJbdUT98qVcc/c6+4DPkl732W90b3n82YjTVYIt52+LDYXaky93fsh2LqpjLvNdClF0TqIUgV8p8txjxgBnw+tsG8WGD1K+X1Q3pqgOLGTHFPwnBTicCVvfFZfY1llCcljIKQdU7HNA1BghCOIEQZ5G4mwkdnRdzeU9hT0iAXrvjZyVJcOmLccHObrdsRjdVwimp90rsb7RPqM5jiTcozB6BqLc/vtFJ6yklnI+Ml8SlpSUzsQsPchewrHjKWF8cSqs8Xsn5yy/SpOEUA4iz2X7PHQHA12zddJiYDX1Sr2e/nY98r/yRzfn+uQAJFaIJZkDHo46MD+7PlkB8+SedoJ+cXNpMN/QktBjatRRdXWgvxQrRfcpUCPx7b3P/2K2o1zzn4dTjtjK0WRCLJvuePnjpDvKUeib67ia7/66T00Lke8pEqGJt6YtgmWsFuSzlxgdTmHSY9tg9l9tRqN5+5lxbdqAXsL4Y863OzALkc3k1jTGTQAv8oqUY0rJQ3eIVjLLgJS7vVaYDE4KlakjjUgZ0F2I/D1aBXCdylpxS0butu4AoJi69DOM1MIxV8SDejJ09oqT5NMKlPUp5fKjv74qhNaqspyXqTWSPVyf3kIKxKu3KecEM8/UTtmIk8kEgbJM7AYyeAcR5DPP2LGXxWU+m5IHmbeiKTiYvzeLzYpGY7HZ01sV1We29+UGvyLaZ4Nyk3x2/hUh6TrkK8a4SbIqzU6KWyh5KZlmqEEUXROC7sQyDnDq/a4JJJN0cp0VR7eYzJNOLF92nwwjnTeW9QTawi1MCiAhlXhZKMTErXo3G9eB2dJSZVFiLmZ2pYuGtpyx1nKL+v7BCrF28lsHRDOe1B1iJbDUEuL8tO5sBG1zSKfP4DbGdupa+GNw0G1yXFASgiilwsPmSe1n5odeRLRurdoKrnoo1Fp51dKiZLZ2zXD1/HV3lEYiuyU4FRheHTpNeNPrW2EbQfTuuM1oekvg+ZFkzQNbOa51x1g9t63QD7wNGGi5RKricKdyal6H3mqF+t01V6/Yb2khstRsE1irVd/dodbKv9t6uHoIai1erlLb0sPTVYtcWgItkvubGFZbUUl3xNbqkpaos6jrJjDXg/rdEVui+21dexE23QBiW7S0h5e3hU1X6OrPIQtbuvpLGDtlXKubEpuPtTZN2qKA5h5iaXVaY4ItjahWN7E24GblSU/IlTqUFdrO2aOsnABZqkOWxbc+VhftcwobUWq9/r+nYi3fBljlWSyPoLTUadk8C9B2U6gtqmNTZUP0On+6tCIbotgtp2WJPAvRbgS4Gbzui9sMYi8LuwyzbgR5LRLdF3wtIr1wDkUkTGOAuazCOENA/vl29iHyXJ5NJ3fgMz5DRTj5yHFHu39zG4LEF0VguRonLpdrWUCvij7KGDSTc9fTko0lET3bOAEkBpzOPmDGZd5I5g61Pu8rr6vf8im6m71KT1V09yzpyLnoP9f6qRj9KZhQIBDvjPZk0ZyT/d55qT7+KX7bqu3uARV2l1W5zAmfIQQTRmB0ldJxEILjeZ7s5uzy+uy3L+8+XZ4fX1cymwO9el9jNk+zmbib2VQVENyRe/CekhTL7ympaLHOq8grzM3JXAF+6NF+nY3zT+ojv/vphx5ujZfKHxhNdCvQ/0UAa2MEedsTgqSXVd61TFYVbvloTlL1N5IjYtcgerG5bguSFzABKm8yESJQ4aNOAjEUDAb4mIZV/BY6ZfN9ffjFwhrMvN93RqNslu9Zq2WXuXZev5CU9l/2pzTazfzQi/6kXr1Sxi3g/dFkdQ954nFkVgzbDKRYNaR53EPwAKhZZnKjnnluqG5PiPoClTwhFMuIsqP9008YybktU2CC4iD/nxTqpSCYUfbVrbLIhrXpoHhjpjbF463trUO16L3J2P4LxrIM0Ukp8nx/yIHYYot+sr4r/Aq4HzImux0Oi06HEbx3YHR0u1WBcrvlhChgTLwZpwhdwT/B7dabw6Fob3RDJIjEieqT8BsomMkm09Gbj9lD3/cPh+K23itfsjWKfwE) proves this is the culprit.

**Comment by OmarShehata** [comment]
@hpinkos if you're more familiar with that class, I can stop there and let you take it from here.

**Comment by hpinkos** [comment]
We have so many problems caused by that `splitLongitude` function.  @OmarShehata keep poking around at it and let me know if you get stuck.  My guess would be something's going weird in this function: https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/GeometryPipeline.js#L1919

**Comment by mramato** [comment]
They released a new minor version update for Chrome last night and it no longer crashes on my system.  Maybe it was a regression on their end?  Can others confirm?

I still maintain that Chrome 71 was garbage, this and #7388 seem to confirm that.

**Comment by OmarShehata** [comment]
@mramato it's still true that `splitLongitude` is spitting out `NaN`'s that are being sent to the GPU in the texture coordinate attribute buffer. If it doesn't crash anymore it might not be as big of a priority but probably still something we should fix.

I'm guessing the crash was not intentionally on Chrome's side because the debug log says something like "Application tried to access memory and was denied" as opposed to "NaN values are not allowed here".

**Comment by hpinkos** [comment]
Re-targeted this issue to fix the texture coordinate computation in `splitLongitude`.  You can see from the gif I posted in the first comment that it's doing something completely crazy.  The numbers its generating aren't within the [0, 1] range.

**Comment by hpinkos** [comment]
Only seems to be a problem with `GroundPrimitive`.  They didn't show up at all prior to 1.49.  We fixed it for 1.49 with #6951 so they display but the texture coordinates are wrong.

**Comment by hpinkos** [comment]
Just kidding, the `GroundPrimitive` texture thing is unrelated to `splitLongitude`.  I'm going to open a separate issue.  We'll keep this one for fixing the NaN thing.
